obs-browser-source.cpp

```diff
@@ -37,6 +37,27 @@ extern bool QueueCEFTask(std::function<void()> task);
 static mutex browser_list_mutex;
 static BrowserSource *first_browser = nullptr;
 
+static void SendBrowserVisibility(CefRefPtr<CefBrowser> browser, bool isVisible)
+{
+	if (!browser)
+		return;
+
+#if ENABLE_WASHIDDEN
+	if (isVisible) {
+		browser->GetHost()->WasHidden(false);
+		browser->GetHost()->Invalidate(PET_VIEW);
+	} else {
+		browser->GetHost()->WasHidden(true);
+	}
+#endif
```

Comment:
Hm, shouldn't it just do an #else for the code below this?  If ENABLE_WASHIDDEN is true shouldn't the emulation be unnecessary?  (I admit I might be misunderstanding)

Reply:
@jp9000 the back end of this sits on top of the OBS Visibility API ("Visibility" message which in turn triggers "onVisibilityChange"), this moves creation and dispatching of the "Visibility" message from SetShowing() into this separate function, and if available, makes use of WasHidden() as well.

Reply:
as a bonus, this simplifies the code in SetShowing()

Reply:
if you move on to browser-app.cpp, you'll notice that actual emulation is indeed disabled when WAS_HIDDEN=1